### PR TITLE
[PWEB-12817] Rails 5.1 Param Safety

### PR DESCRIPTION
**Pull request by HoyaBoya**
# About

This applies a Rails 5.1 PR from the Virtus gem into our fork, along a `branch/1_0_5` which we will maintain for long running changes. Hopefully this all gets deprecated soon.

Once this is ready, I will point Babylon to this branch of the fork.

## Related PRs

https://github.com/solnic/virtus/pull/388

## Related JIRA

https://perchwell.atlassian.net/issues/PWEB-12817?jql=ORDER%20BY%20created%20DESC

### CODE REVIEW

File: lib/virtus/attribute_set.rb [hunk 1]
```diff
@@ -193,9 +193,13 @@ def set_defaults(object, filter = method(:skip_default?))
     #
     # @api private
     def coerce(attributes)
-      ::Hash.try_convert(attributes) or raise(
-        NoMethodError, "Expected #{attributes.inspect} to respond to #to_hash"
-      )
+      if attributes.respond_to?(:to_unsafe_hash)
```
Comment (HoyaBoya) [inline lib/virtus/attribute_set.rb:196]: Not a Rails expert, but I think this feels ok, I probably would have used an inheritance call of `instance_of?(Hash)` but perhaps this is more Ruby-esque.

**Comment by HoyaBoya** [comment]
Any thoughts on this before I merge and point Babylon to this fork? @jonmoniaci-perch @rhaus @dmartin-perchwell

**Review by jonmoniaci-perch — APPROVED** [review]
Given the state of things in Babylon, I think this commit is fine.

Ideally we would _never_ pass controller `params` deeper than maybe a service object.  And therefore would never even have to fork this gem... But, unfortunately that's what Babylon engineers have done over time and this is a "single touch point" to ease the pain.

**Comment by HoyaBoya** [comment]
> Ideally we would never pass controller params deeper than maybe a service object. And therefore would never even have to fork this gem... But, unfortunately that's what Babylon engineers have done over time and this is a "single touch point" to ease the pain.

Agreed on so many levels.

Also add, "Babylon engineers should not let things go stale"... to the 10 Commandments of Perchwell Engineering we need.

**Pull request merged**